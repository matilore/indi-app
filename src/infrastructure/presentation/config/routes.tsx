import { createBrowserRouter } from "react-router-dom";
import { Home, PodcastDetails } from "../pages";
import { ROUTES_PATHS } from "./routesConstants";
import { Layout } from "../layout";

export const routesConfig = [
  {
    path: ROUTES_PATHS.HOME,
    element: <Layout />,
    children: [
      {
        path: ROUTES_PATHS.HOME,
        element: <Home />,
        loadingMode: "render",
      },
      {
        path: ROUTES_PATHS.PODCAST_DETAILS,
        element: <PodcastDetails />,
        loadingMode: "render",
      },
    ],
  },
];

export const router = createBrowserRouter(routesConfig);
