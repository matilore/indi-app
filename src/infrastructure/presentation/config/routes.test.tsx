import { routesConfig } from "./routes";

describe("Router", () => {
  test("should create a router with the correct configuration", () => {
    expect(routesConfig).toMatchSnapshot();
  });
});
