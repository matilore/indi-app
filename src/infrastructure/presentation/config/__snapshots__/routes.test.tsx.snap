// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Router > should create a router with the correct configuration 1`] = `
[
  {
    "children": [
      {
        "element": <Home />,
        "path": "/",
      },
      {
        "element": <PodcastDetails />,
        "path": "/podcast/:id",
      },
    ],
    "element": <Layout />,
    "path": "/",
  },
]
`;
